# Breast Cancer  분류해보기 실습

- 필요한 모듈 Import 하기

In [5]:
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

- 데이터 준비하기

In [28]:

from sklearn import preprocessing
import numpy as np

from sklearn.tree import DecisionTreeClassifier
decision_tree = DecisionTreeClassifier(random_state=32)

from sklearn.ensemble import RandomForestClassifier
random_forest = RandomForestClassifier(random_state=32)

from sklearn import svm
svm_model = svm.SVC()

from sklearn.linear_model import SGDClassifier
sgd_model = SGDClassifier()

from sklearn.linear_model import LogisticRegression
logistic_model = LogisticRegression()

def classfier(load_data):
    all_data=load_data
    data_data=all_data.data
    data_label=all_data.target
    
    X_train,X_test,y_train,y_test=train_test_split(data_data,
                                               data_label,
                                               test_size=0.2,
                                               random_state=70)
    

    X_train = preprocessing.scale(X_train)
    X_test = preprocessing.scale(X_test)
    
    
        
    models=[DecisionTreeClassifier(random_state=32),
            RandomForestClassifier(random_state=32),
                                    svm.SVC(),
                                    SGDClassifier(),
                                    LogisticRegression()
                                    ]
    
    for model in models:
                                    
        model.fit(X_train, y_train)
        model_predict=model.predict(X_test)
        print(classification_report(y_test,model_predict))

        
a=classfier(load_breast_cancer())
print(a)

              precision    recall  f1-score   support

           0       0.93      0.93      0.93        43
           1       0.96      0.96      0.96        71

    accuracy                           0.95       114
   macro avg       0.94      0.94      0.94       114
weighted avg       0.95      0.95      0.95       114

              precision    recall  f1-score   support

           0       0.95      0.95      0.95        43
           1       0.97      0.97      0.97        71

    accuracy                           0.96       114
   macro avg       0.96      0.96      0.96       114
weighted avg       0.96      0.96      0.96       114

              precision    recall  f1-score   support

           0       0.93      0.98      0.95        43
           1       0.99      0.96      0.97        71

    accuracy                           0.96       114
   macro avg       0.96      0.97      0.96       114
weighted avg       0.97      0.96      0.97       114

              preci

In [11]:
patients = load_breast_cancer()
patients_data=patients.data
patients_label=patients.target

 - 데이터 이해하기
 1. feature data 지정하기
 2. label data 지정하기
 3. Target Names 출력해보기
 4. 데이터 Describe 해보기

In [13]:
print("\n데이터의 키값 \n", patients.keys())

print("\n데이터 모양 \n","데이터 갯수 : ",patients.data.shape[0],"\n데이터 크기 : ", patients.data.shape[1])

print("데이터 셈플\n", patients_data[0])

print(patients.feature_names)
print("\n 타켓 이름 : \n", patients.target_names)


데이터의 키값 
 dict_keys(['data', 'target', 'target_names', 'DESCR', 'feature_names', 'filename'])

데이터 모양 
 데이터 갯수 :  569 
데이터 크기 :  30
데이터 셈플
 [1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
 1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
 6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
 1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
 4.601e-01 1.189e-01]
['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension'

In [14]:

print("데이터 값 : \n", patients.data[0])


데이터 값 : 
 [1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
 1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
 6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
 1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
 4.601e-01 1.189e-01]


In [15]:
print(patients_label)
print(patients.target_names)

[0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 1 0 0 0 0 0 0 0 0 1 0 1 1 1 1 1 0 0 1 0 0 1 1 1 1 0 1 0 0 1 1 1 1 0 1 0 0
 1 0 1 0 0 1 1 1 0 0 1 0 0 0 1 1 1 0 1 1 0 0 1 1 1 0 0 1 1 1 1 0 1 1 0 1 1
 1 1 1 1 1 1 0 0 0 1 0 0 1 1 1 0 0 1 0 1 0 0 1 0 0 1 1 0 1 1 0 1 1 1 1 0 1
 1 1 1 1 1 1 1 1 0 1 1 1 1 0 0 1 0 1 1 0 0 1 1 0 0 1 1 1 1 0 1 1 0 0 0 1 0
 1 0 1 1 1 0 1 1 0 0 1 0 0 0 0 1 0 0 0 1 0 1 0 1 1 0 1 0 0 0 0 1 1 0 0 1 1
 1 0 1 1 1 1 1 0 0 1 1 0 1 1 0 0 1 0 1 1 1 1 0 1 1 1 1 1 0 1 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 1 1 1 1 1 1 0 1 0 1 1 0 1 1 0 1 0 0 1 1 1 1 1 1 1 1 1 1 1 1
 1 0 1 1 0 1 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1 1 1 0 1 0 1 1 1 1 0 0 0 1 1
 1 1 0 1 0 1 0 1 1 1 0 1 1 1 1 1 1 1 0 0 0 1 1 1 1 1 1 1 1 1 1 1 0 0 1 0 0
 0 1 0 0 1 1 1 1 1 0 1 1 1 1 1 0 1 1 1 0 1 1 0 0 1 1 1 1 1 1 0 1 1 1 1 1 1
 1 0 1 1 1 1 1 0 1 1 0 1 1 1 1 1 1 1 1 1 1 1 1 0 1 0 0 1 0 1 1 1 1 1 0 1 1
 0 1 0 1 1 0 1 0 1 1 1 1 1 1 1 1 0 0 1 1 1 1 1 1 0 1 1 1 1 1 1 1 1 1 1 0 1
 1 1 1 1 1 1 0 1 0 1 1 0 

 - 데이터셋 분류하기

In [16]:
X_train,X_test,y_train,y_test=train_test_split(patients_data,
                                               patients_label,
                                               test_size=0.2,
                                               random_state=70)
from sklearn import preprocessing
import numpy as np
X_train = preprocessing.scale(X_train)
X_test = preprocessing.scale(X_test)

 - 모델 불러오기

In [18]:
from sklearn.tree import DecisionTreeClassifier
decision_tree = DecisionTreeClassifier(random_state=32)

from sklearn.ensemble import RandomForestClassifier
random_forest = RandomForestClassifier(random_state=32)

from sklearn import svm
svm_model = svm.SVC()

from sklearn.linear_model import SGDClassifier
sgd_model = SGDClassifier()

from sklearn.linear_model import LogisticRegression
logistic_model = LogisticRegression()

accuracy_model={}

In [19]:
#Decision Tree
decision_tree.fit(X_train, y_train)
decision_y_predict=decision_tree.predict(X_test)
char_Decision=classification_report(y_test,decision_y_predict)
print(char_Decision)

              precision    recall  f1-score   support

           0       0.93      0.93      0.93        43
           1       0.96      0.96      0.96        71

    accuracy                           0.95       114
   macro avg       0.94      0.94      0.94       114
weighted avg       0.95      0.95      0.95       114



In [20]:
#RandomForest
random_forest.fit(X_train, y_train)
random_forest_y_predict=random_forest.predict(X_test)
char_Random=classification_report(y_test,random_forest_y_predict)
print(char_Random)

              precision    recall  f1-score   support

           0       0.95      0.95      0.95        43
           1       0.97      0.97      0.97        71

    accuracy                           0.96       114
   macro avg       0.96      0.96      0.96       114
weighted avg       0.96      0.96      0.96       114



In [21]:
#SVM
svm_model.fit(X_train, y_train)
svm_model_y_predict=svm_model.predict(X_test)
char_SVM=classification_report(y_test,decision_y_predict)
print(char_SVM)

              precision    recall  f1-score   support

           0       0.93      0.93      0.93        43
           1       0.96      0.96      0.96        71

    accuracy                           0.95       114
   macro avg       0.94      0.94      0.94       114
weighted avg       0.95      0.95      0.95       114



In [22]:
#SGD
sgd_model.fit(X_train, y_train)
sgd_model_y_predict=sgd_model.predict(X_test)
char_SGD=classification_report(y_test,sgd_model_y_predict)
print(classification_report(y_test,sgd_model_y_predict))

              precision    recall  f1-score   support

           0       0.89      0.91      0.90        43
           1       0.94      0.93      0.94        71

    accuracy                           0.92       114
   macro avg       0.91      0.92      0.92       114
weighted avg       0.92      0.92      0.92       114



In [23]:
#logistic Regression
logistic_model.fit(X_train, y_train)
logistic_model_y_predict=logistic_model.predict(X_test)
char_Logistic=classification_report(y_test,sgd_model_y_predict)
print(char_Logistic)

              precision    recall  f1-score   support

           0       0.89      0.91      0.90        43
           1       0.94      0.93      0.94        71

    accuracy                           0.92       114
   macro avg       0.91      0.92      0.92       114
weighted avg       0.92      0.92      0.92       114



In [24]:

print(accuracy_model)

{}


## Breast_Cancer 데이터셋 학습 결과
## Recall 값 비교 결과, RANDOM FOREST의 정확도가 96% 로 가장 좋다.